libcontainer/cgroups/systemd/v1.go

```diff
@@ -225,6 +229,10 @@ func (m *legacyManager) Destroy() error {
 	unitName := getUnitName(m.cgroups)
 
 	stopErr := stopUnit(dbusConnection, unitName)
+	// reset it if Dbus is closed
+	if isDbusClosed(stopErr) {
+		m.dbus.resetDbusConn()
```

Comment:
So, as I asked before -- do we want to retry in such case?

Reply:
Same question to similar cases below

Reply:
Call to `getConnection` will retry, i think this should be enough.

Reply:
So yes, this is definitely an improvement as we're no longer stuck in "dbus: connection closed by user" state.

I think this may be improved further if any operation that requires dbus and receives this error will not only close the connection, but also retry (i.e. get a new connection, retry the operation).

Maybe it makes sense to add a method to dbusConnManager which will execute a function, check if the error it returns is "dbus closed" and if it is, close, reconnect and retry. This way we can wrap all the calls into this (rather than modify the code in a bunch of places to check for this particular error and call close).

Reply:
Thanks. updated.